### ISSUE TITLE
Updating variable that is used for initialDate int the CalendarDatePicker widget does not update the selected date

### ISSUE THREAD

**Issue by GauteHaugen** [flutter/flutter#60981]
## Steps to Reproduce
1. Create a new flutter project
2. Replace the template main.dart with code below.

**Expected results:**
When clicking the arrow buttons, I wanted the date variable to update and use the setState method to rebuild the CalendarDatePicker with the new date as the selected date.

**Actual results:**
The CalendarDatePicker gets redrawn, but the selected date does not change. The date variable does change as the Text widget under the buttons gets updated with the new date value.

Does the CalendarDatePicker widget not support changing the selected date using anything else then user input directly on the widget itself? My use case would be to have some predefined dates under the calendar that the user could click on instead of the calendar, and the calendar should be updated to have the selected date be the same as the predefined date.

<details>
[√] Flutter (Channel stable, v1.17.5, on Microsoft Windows [Version 10.0.18362.900], locale en-US)
    • Flutter version 1.17.5 at C:\flutter
    • Framework revision 8af6b2f038 (7 days ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\gaute.haugen\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.46.1)
    • VS Code at C:\Users\gaute.haugen\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.12.1

[√] Connected device (1 available)
    • SM G975F • android-arm64 • Android 10 (API 29)

• No issues found!
</details>

**Code:**
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  var date = DateTime.now().toUtc();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Column(
        children: <Widget>[
          CalendarDatePicker(
            initialDate: date ?? DateTime.now(),
            firstDate: DateTime(1970, 1, 1),
            lastDate: DateTime(2099, 31, 12),
            onDateChanged: (DateTime value) {  },
          ),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: <Widget>[
              IconButton(
                icon: Icon(Icons.arrow_back),
                onPressed: () {
                  setState(() {
                    date = date.add(Duration(days: -1));
                  });
                },
              ),
              IconButton(
                icon: Icon(Icons.arrow_forward),
                onPressed: () {
                  setState(() {
                    date = date.add(Duration(days: 1));
                  });
                },
              )
            ],
          ),
          Text('Current date: ${date.toString()}'),
        ],
      )
    );
  }
}

```

**Comment by TahaTesser** [comment]
Hi @GauteHaugen 
Adding ` currentDate: date ?? DateTime.now()`changes current in your code sample but i think you blue circle selection to change. Please confirm that's the behaviour you are looking

<details> 
<summary>Code Sample</summary> 

``` 
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  var date = DateTime.now().toUtc();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(widget.title),
        ),
        body: Column(
          children: <Widget>[
            CalendarDatePicker(
              initialDate: date ?? DateTime.now(),
              firstDate: DateTime(1970, 1, 1),
              lastDate: DateTime(2099, 31, 12),
              currentDate: date ?? DateTime.now(),
              onDateChanged: (DateTime value) {},
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: <Widget>[
                IconButton(
                  icon: Icon(Icons.arrow_back),
                  onPressed: () {
                    setState(() {
                      date = date.add(Duration(days: -1));
                    });
                  },
                ),
                IconButton(
                  icon: Icon(Icons.arrow_forward),
                  onPressed: () {
                    setState(() {
                      date = date.add(Duration(days: 1));
                    });
                  },
                )
              ],
            ),
            Text('Current date: ${date.toString()}'),
          ],
        ));
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel dev, 1.20.0-3.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-3.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision 0af027f805 (3 days ago), 2020-07-04 12:19:20 -0700
    • Engine revision a751393804
    • Dart version 2.9.0 (build 2.9.0-20.0.dev 22da8934ac)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (4 available)
    • SM M305F (mobile)      • 32003c30dc19668f          • android-arm64  •
      Android 10 (API 29)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      13.5.1
    • Web Server (web)       • web-server                • web-javascript •
      Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 83.0.4103.116

• No issues found!

``` 

</details> 

Thank you

**Comment by GauteHaugen** [comment]
I can confirm that this works in dev channel, 1.20.0-3.0.pre, but not on the latest stable release. Do you know if this feature will be included in the next stable release?

**Comment by TahaTesser** [comment]
Hi @GauteHaugen 
Yes, it will be  get to `stable` channel from dev to beta. Since its not reproducible in `dev` channel
given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you

**Comment by GauteHaugen** [comment]
Hi, this feature has now gone to the release channel. But the feature does not work the way I expected or the way I verified it working in dev. In the calendar picker, there are two circles, one to indicate which day is the current day and one circle to indicate the selected date. I am now able to change the current date in the calendar using code, but I'm still unable to change the currently selected date which is what I want to.

Looking at the code, the private variable _selectedDate is only set in the initState function and in the private function _handleDayChanged. This is in the file calendar_date_picker.dart on line 169 and 245.

I'm still using the same example I provided in the initial question.
